crates/runnable/src/static_runnable.rs

```diff
@@ -0,0 +1,48 @@
+//! Definitions of runnables with a static file config definition, not dependent on the application state.
+
+use std::path::{Path, PathBuf};
+
+use crate::{static_source::Definition, Runnable, RunnableId, SpawnInTerminal};
+
+/// A single config file entry with the deserialized runnable definition.
+#[derive(Clone, Debug, PartialEq)]
+pub struct StaticRunnable {
+    id: RunnableId,
+    definition: Definition,
+}
+
+impl StaticRunnable {
+    pub(super) fn new(id: usize, runnable: Definition) -> Self {
+        Self {
+            id: RunnableId(format!("static_{}_{}", runnable.label, id)),
```

Comment:
Nit: we should use EntityId of a spawning Source as part of the identifier, as otherwise you two different `StaticSource`s will collide on IDs here. Let's take a `WeakModel<Source>` as an argument here.

Reply:
On the other hand, given how much headache https://github.com/zed-industries/zed/pull/8009/commits/e95e18cc3d5e39c5fee06f475263dd6902b3c517 was, maybe let's not use entity id here.. =)

Reply:
Yeah, I did try using entities here initially, but that old code used to call `new_model` every time we queried the runnables list — so effectively the runnables were re-created with different IDs every time, spoiling the entire terminal tab reuse logic.

Let's keep this in mind when working with dynamic runnables and change that, if needed.